File: PatientService/PatientService/PatientService.Db/Entities/Illness.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace PatientService.Db.Entities
{
    public enum IllnessType 
    {
        Allergies,
        ColdOrFlu,
        Conjunctivitis,
        Diarrhea,
        Headaches,
        Mononucleosis,
        StomachAches,
        Virus
    }
    public class Illness
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string IllName { get; set; }
        [Required]
        public IllnessType Type { get; set; }
        public virtual List<Patient> Patients { get; set; }
    }
}

File: PatientService/PatientService/PatientService.WebClient/ClientApp/src/app/illness/Illness.ts
import {Patient} from '../patient/Patient';

export class Illness {
  constructor(
    public  id?: number,
    public  illName?: string,
    public illType?: illnessType,
    public patients?: Array<Patient>
  ) {}
}

enum illnessType {
  Allergies,
  ColdOrFlu,
  Conjunctivitis,
  Diarrhea,
  Headaches,
  Mononucleosis,
  StomachAches,
  Virus
}
